<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.56">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>2&nbsp; Experimental design and climate condition – PhD soybean 2021-2024</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<link href="../../report/water/2_water_during_the_experiment.html" rel="next">
<link href="../../report/intro/intro.html" rel="prev">
<link href="../../src/root.png" rel="icon" type="image/png">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script src="../../site_libs/quarto-contrib/iconify-2.1.0/iconify-icon.min.js"></script>
<script src="../../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../report/intro/intro.html"><strong>Introduction to the thesis</strong></a></li><li class="breadcrumb-item"><a href="../../report/intro/1_experimental_design_climat.html"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Experimental design and climate condition</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="../../index.html" class="sidebar-logo-link">
      <img src="../../src/root.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
    <div class="sidebar-title mb-0 py-0">
      <a href="../../">PhD soybean 2021-2024</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/cmaslard/phd_soybean_2021_2024/" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
    <a href="../../guide-R.pdf" title="Download PDF" class="quarto-navigation-tool px-1" aria-label="Download PDF"><i class="bi bi-file-pdf"></i></a>
    <div class="dropdown">
      <a href="" title="Share" id="quarto-navigation-tool-dropdown-0" class="quarto-navigation-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false" role="link" aria-label="Share"><i class="bi bi-share"></i></a>
      <ul class="dropdown-menu" aria-labelledby="quarto-navigation-tool-dropdown-0">
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="https://twitter.com/intent/tweet?url=|url|">
              <i class="bi bi-twitter pe-1"></i>
            Twitter
            </a>
          </li>
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="https://www.facebook.com/sharer/sharer.php?u=|url|">
              <i class="bi bi-facebook pe-1"></i>
            Facebook
            </a>
          </li>
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="https://www.linkedin.com/sharing/share-offsite/?url=|url|">
              <i class="bi bi-linkedin pe-1"></i>
            LinkedIn
            </a>
          </li>
      </ul>
    </div>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
    <div id="quarto-search" class="quarto-navigation-tool px-1" title="Search"></div>
</div>
    </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preface</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text"><strong>Introduction to the thesis</strong></span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../report/intro/intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../report/intro/1_experimental_design_climat.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Experimental design and climate condition</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text"><strong>🌱 Ecophysiology</strong></span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../report/water/2_water_during_the_experiment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Use of water during the experiment</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../report/physio/2_RUE.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Calcul of Radiation Use Efficiency</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../report/physio/0_preprocessing_ecophy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Pre-processing of ecophysiological data</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../report/physio/2_rapid_analysis_ecophy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Rapid analysis of ecophysiological data</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false">
 <span class="menu-text"><strong>🫚 Root architecture</strong></span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../report/root_architecture/0_preprocessing_root_architecture.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Pre-processing of root architecture data</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../report/root_architecture/2_rapid_analysis_root_architecture.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Rapid analysis of root architecture data</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../report/root_architecture/2_root_architecturedepending_depth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Analyse Root Architecture dependent of the depth</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text"><strong>⚛️ Ionomics</strong></span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../report/iono/0A_preprocessing_iono.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Pre-processing of ionomic data</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../report/iono/0B_preprocessing_efficiency.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Pre-processing for efficiency in ionomic analyse</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../report/iono/2_rapid_analysis_iono.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Analysis of ionome</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false">
 <span class="menu-text"><strong>📊 Multi-Omics Integration</strong></span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../report/multi_omics/A1_physio_iono_RA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Root architecture - Ionomics - Biomass - Water - Licor</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#experimental-design" id="toc-experimental-design" class="nav-link active" data-scroll-target="#experimental-design"><span class="header-section-number">2.1</span> Experimental design</a></li>
  <li><a href="#climate-in-the-greenhouse" id="toc-climate-in-the-greenhouse" class="nav-link" data-scroll-target="#climate-in-the-greenhouse"><span class="header-section-number">2.2</span> Climate in the greenhouse</a>
  <ul class="collapse">
  <li><a href="#correction-for-spatial-trends-over-time" id="toc-correction-for-spatial-trends-over-time" class="nav-link" data-scroll-target="#correction-for-spatial-trends-over-time"><span class="header-section-number">2.2.1</span> Correction for spatial trends over time</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/cmaslard/phd_soybean_2021_2024/blob/main/report/intro/1_experimental_design_climat.qmd" class="toc-action"><i class="bi bi-github"></i>View source</a></li><li><a href="https://github.com/cmaslard/phd_soybean_2021_2024/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../report/intro/intro.html"><strong>Introduction to the thesis</strong></a></li><li class="breadcrumb-item"><a href="../../report/intro/1_experimental_design_climat.html"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Experimental design and climate condition</span></a></li></ol></nav>
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Experimental design and climate condition</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">#pkg </span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(statgenHTP) <span class="co">#remotes::install_github("Biometris/statgenHTP", ref = "develop", dependencies = TRUE)</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SpATS)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(lubridate)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gridExtra)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="co"># src </span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"src/function/card_compilation.R"</span>))</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="co"># cosmetics</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>temperature_color<span class="ot">=</span><span class="fu">read_excel</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/color_palette.xlsm"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(set <span class="sc">==</span> <span class="st">"greenhouse"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(color, treatment) <span class="sc">%&gt;%</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>      <span class="fu">pull</span>(color) <span class="sc">%&gt;%</span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>      <span class="fu">setNames</span>(<span class="fu">read_excel</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/color_palette.xlsm"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">filter</span>(set <span class="sc">==</span> <span class="st">"greenhouse"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">pull</span>(treatment)</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>               )</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>my_theme<span class="ot">=</span><span class="fu">theme_classic</span>()<span class="sc">+</span><span class="fu">theme</span>(</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>, <span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>  <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>  <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>  <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>()</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="experimental-design" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="experimental-design"><span class="header-section-number">2.1</span> Experimental design</h2>
<p>Two French genotypes of soybean (Glycine max L.) belonging to the same maturity group I were selected for their similar root or aerial biomasses and their contrasted root system architecture: Stocata (obtained from RAGT Semences) and Wendy (obtained from Caussade Semences) <span class="citation" data-cites="maslard2021">(<a href="../../references.html#ref-maslard2021" role="doc-biblioref">Maslard et al. 2021</a>)</span>. Seeds were calibrated and pre-germinated for 4 days at 20 °C in a Fitoclima S600 germinator (Aralab, Rio de Mouro, Portugal) before transplantation. Seedlings were grown in RhizoTubes® (<a href="../root_architecture/0_preprocessing_root_architecture.html#fig-rhizotube" class="quarto-xref">Figure&nbsp;<span>7.1</span></a>) filled with sandy soil, allowing the visualization of the root system (<a href="../root_architecture/0_preprocessing_root_architecture.html" class="quarto-xref"><span>Chapter 7</span></a>). The soil was collected from a grassland at Flammerans near Dijon (Burgundy). Plants were grown under controlled conditions, in the Plant Phenotyping Platform for Plant and Microorganism Interaction (4PMI) at INRAE in Dijon (France) (47°32’N, 5°02’E) during 23 days in RhizoTube® (<a href="#fig-experimantal_design" class="quarto-xref">Figure&nbsp;<span>2.1</span></a>). Two seedlings were transplanted in each of the 144 RhizoTubes®, in two different identical greenhouses (<a href="#fig-greenhouse" class="quarto-xref">Figure&nbsp;<span>2.4</span></a>). Two plants grew in each RhizoTubes® and were destined for four different types of analysis (<a href="#fig-harvest_design" class="quarto-xref">Figure&nbsp;<span>2.2</span></a>). At the same time, each seedling was inoculated with 1mL of <em>Bradyrhizobium diazoefficiens</em> corresponding to 108 rhizobia.</p>
<div id="fig-experimantal_design" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-experimantal_design-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="img/Experimental_design.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1" title="Figure&nbsp;2.1: Experimental design of the experiment"><img src="img/Experimental_design.png" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-experimantal_design-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2.1: Experimental design of the experiment
</figcaption>
</figure>
</div>
<div id="fig-harvest_design" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-harvest_design-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="img/harvest_design.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2" title="Figure&nbsp;2.2: Harvest diagram for the experiment. RT for Rhizotube. Harvest 1 took place 5 days after sowing. Harvest 2 was carried out on the day after sowing. For harvest 1, there are two plants of the same genotype per rhizotube (RT0). For each plant, ecophysiological measurements and the content of elements in each compartment were conducted. For harvest 2, in each rhizotube (RT1, RT2), there were two plants of the same genotype that have undergone the same stress conditions. Ecophysiological measurements and the element content were performed on one of the two plants. The other plant was designated for molecular analyses. Transcriptomic measurements were conducted on a pool of two roots from two plants from two different RhizoTubes (RT1 and RT2) of the same genotype having received the same stress conditions. Metabolomic analyses of leaves, stems, and roots were performed on the same plant combination. The soil near this plant was collected for metabolomic analyses, pH analyses, and microbial community analyses (Rhizosphere) (no assembly of these soils). The soil between these two plants was used for metabolomic analyses as well as for microbial community analyses (Bulk soil). During this experiment, other RhizoTubes (RT3) also containing two plants of the same genotype were in the same greenhouse. One was designated for microbial community analyses, the other for metabolomic measurements of exudates. On the plant used for the root exudates, ecophysiological measurements were also conducted. Root system architecture (RSA) and total evapotranspiration were measured on all RhizoTubes during the experiment. WW: Well Watered; WS: Water Stress; OT: Optimal temperature; HS: Heat Stress."><img src="img/harvest_design.png" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-harvest_design-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2.2: Harvest diagram for the experiment. RT for Rhizotube. Harvest 1 took place 5 days after sowing. Harvest 2 was carried out on the day after sowing. For harvest 1, there are two plants of the same genotype per rhizotube (RT0). For each plant, ecophysiological measurements and the content of elements in each compartment were conducted. For harvest 2, in each rhizotube (RT1, RT2), there were two plants of the same genotype that have undergone the same stress conditions. Ecophysiological measurements and the element content were performed on one of the two plants. The other plant was designated for molecular analyses. Transcriptomic measurements were conducted on a pool of two roots from two plants from two different RhizoTubes (RT1 and RT2) of the same genotype having received the same stress conditions. Metabolomic analyses of leaves, stems, and roots were performed on the same plant combination. The soil near this plant was collected for metabolomic analyses, pH analyses, and microbial community analyses (Rhizosphere) (no assembly of these soils). The soil between these two plants was used for metabolomic analyses as well as for microbial community analyses (Bulk soil). During this experiment, other RhizoTubes (RT3) also containing two plants of the same genotype were in the same greenhouse. One was designated for microbial community analyses, the other for metabolomic measurements of exudates. On the plant used for the root exudates, ecophysiological measurements were also conducted. Root system architecture (RSA) and total evapotranspiration were measured on all RhizoTubes during the experiment. WW: Well Watered; WS: Water Stress; OT: Optimal temperature; HS: Heat Stress.
</figcaption>
</figure>
</div>
<p>Air relative humidity was controlled to 45% and the photoperiod was set to 16h thanks to an artificial lighting (PAR of 272µmol.m-2. s-1) supplied with sodium lamps (400W lamp, HPS Plantastar, OSRAM, Munich, Germany). The PAR data as well as the other climatic data (humidity, temperature) are presented below (<a href="#fig-metadata" class="quarto-xref">Figure&nbsp;<span>2.3</span></a>).</p>
<p>Every Rhizotube® was autonomously weighed both before and after each watering session, if applicable, three times a day. This process was conducted to establish the evolving water evapotranspiration (ml)(<a href="../water/2_water_during_the_experiment.html#fig-evapotranspiration" class="quarto-xref">Figure&nbsp;<span>3.2</span></a>) and precise solution quantities (<a href="../water/2_water_during_the_experiment.html" class="quarto-xref"><span>Chapter 3</span></a>). The cumulative evapotranspiration during experiment was computed by summing and halving (accounting for the two plants within each Rhizotube®) the individual daily differences in Rhizotube® weight between consecutive watering events, or weight variation (<a href="../water/2_water_during_the_experiment.html" class="quarto-xref"><span>Chapter 3</span></a>). A N-free nutrient solution as described in <span class="citation" data-cites="voisin_symbiotic_2003">(<a href="../../references.html#ref-voisin_symbiotic_2003" role="doc-biblioref">Voisin et al. 2003</a>)</span> was applied 3 times a day to allow 20% SWC (80% of field capacity) of water content in the RhizoTube® (<a href="../water/2_water_during_the_experiment.html#fig-water_content" class="quarto-xref">Figure&nbsp;<span>3.1</span></a>). Mean day/night temperatures were 32°C/24°C in two greenhouses (<a href="#fig-metadata" class="quarto-xref">Figure&nbsp;<span>2.3</span></a>).</p>
<p>Following the first week of growth, 12 plants per genotype were harvested (<a href="#fig-harvest_design" class="quarto-xref">Figure&nbsp;<span>2.2</span></a>). Subsequently, we implemented a factorial design consisting of eight different conditions (Sto_WW_OT, Sto_WS_OT, Sto_WW_HS, Sto_WS_HS, Wen_WW_OT, Wen_WS_OT, Wen_WW_HS, Wen_WS_HS) involving the two genotypes, Stocata (Sto) and Wendy (Wen), and four climatic conditions, composed of two stress factors. One factor was related to watering, with half of plants being subjected to either well-watered (WW) conditions with a SWC of 20 percent (field capacity of 80%) soil water content of 20%, or water stress (WS) conditions achieved by withholding water to reach 10% soil water content (<a href="../water/2_water_during_the_experiment.html#fig-water_content" class="quarto-xref">Figure&nbsp;<span>3.1</span></a>). The second factor was related to temperature. One greenhouse was maintained at an optimal temperature (OT) of 32°C/24°C (day/night), while the other greenhouse experienced two progressive heat waves of 40°C for 3 days (<a href="#fig-metadata" class="quarto-xref">Figure&nbsp;<span>2.3</span></a> A), representing heat stress conditions (HS). Five days after sowing some plants were harvested for physiological analysis (including ionomic analysis) (H1). Then 20 days after sowing, (after the period of stress) plant were harvest (H2) and each organ was collected separately (leaves, stems and roots).</p>
<p><strong>Four different types of analysis were performed:</strong></p>
<ol type="A">
<li>Eight plants per condition were dedicated to ecophysiological analyses (<a href="../water/2_water_during_the_experiment.html" class="quarto-xref"><span>Chapter 3</span></a>) including ionomics (<a href="../iono/0A_preprocessing_iono.html" class="quarto-xref"><span>Chapter 10</span></a>) (8 plants per condition (only 6 plants over the 8 plants were used for ionomic analysis).</li>
<li>Eight plants per condition were dedicated to transcriptomic and metabolomic analyses. After being harvested in liquid nitrogen, four replicates of two plants each were pooled together for each of the four treatments, for each genotype and each plant compartment (leaf, stem, and root). These pooled samples were then ground for analysis. Only root samples were analysed for transcriptomic analysis (<span class="quarto-unresolved-ref">?sec-rnaseq</span>) and all three compartments were analyzed for metabolomics (<span class="quarto-unresolved-ref">?sec-metabo</span>). Plant soil located close to or far from the roots was collected for metabolomics (<span class="quarto-unresolved-ref">?sec-metabo</span>) analyses as well as for microbial community (<span class="quarto-unresolved-ref">?sec-mcom</span>) and pH analyses.</li>
<li>Five plants per condition were dedicated to exudate collection and exudation analyses by metabolomics (<span class="quarto-unresolved-ref">?sec-metabo</span>) and to biomass measurement of each compartment.</li>
<li>Five plants per condition were dedicated to the analysis of microbial communities (fungi or bacteria) in different compartments (phylloplane, leaf endosphere, root endosphere, and rhizoplane) (<span class="quarto-unresolved-ref">?sec-mcom</span>)</li>
</ol>
<p>Plants A and B were harvested from the same rhizotube. Plants C and D were harvested from the same rhizotube. All plants were imaged three times a week to measure aerial and root architecture.</p>
</section>
<section id="climate-in-the-greenhouse" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="climate-in-the-greenhouse"><span class="header-section-number">2.2</span> Climate in the greenhouse</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co">#temperature compilation</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>data_global_climat_select<span class="ot">=</span><span class="fu">climat_function</span>()</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>data_global_climat_select_moy<span class="ot">=</span><span class="fu">mean_interval_function</span>(data_global_climat_select,<span class="dv">5</span>,<span class="st">"min"</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co">#graphique</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplotly</span>(<span class="fu">ggplot</span>(data_global_climat_select_moy,<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">as.POSIXct</span>(Date_Time),<span class="at">y=</span>Hygrometry,<span class="at">col=</span>CardID))<span class="sc">+</span><span class="fu">geom_line</span>()<span class="sc">+</span><span class="co">#geom_point(size=0.5)+</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>           <span class="fu">facet_grid</span>(<span class="sc">~</span>Unit))</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Specify cutting interval in minutes (e.g. 60 minutes)</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>interval_minutes <span class="ot">&lt;-</span> <span class="dv">60</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>data_mean <span class="ot">&lt;-</span> data_global_climat_select_moy <span class="sc">%&gt;%</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(CardID<span class="sc">!=</span><span class="st">"ARIA"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Date_Time =</span><span class="fu">as.POSIXct</span>(Date_Time, <span class="at">format=</span><span class="st">"%Y-%m-%d %H:%M"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Date_Time<span class="sc">&gt;</span><span class="fu">as.POSIXct</span>(<span class="st">"2021-09-22 10:00:00"</span>)<span class="sc">&amp;</span> Date_Time<span class="sc">&lt;</span><span class="fu">as.POSIXct</span>(<span class="st">"2021-10-14 00:00:00"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">interval_id =</span> <span class="fu">cut</span>(</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.POSIXct</span>(Date_Time),</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="fu">min</span>(Date_Time), <span class="fu">max</span>(Date_Time) <span class="sc">+</span> interval_minutes <span class="sc">*</span> <span class="dv">60</span>, <span class="at">by =</span> interval_minutes <span class="sc">*</span> <span class="dv">60</span>),</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">include.lowest =</span> <span class="cn">TRUE</span></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span> </span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(Unit, interval_id) <span class="sc">%&gt;%</span></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">mean_temp =</span> <span class="fu">mean</span>(Temperature, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>            <span class="at">mean_humidity =</span> <span class="fu">mean</span>(Hygrometry, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>            <span class="at">mean_PAR =</span> <span class="fu">mean</span>(PAR, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() </span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>plot_mean_temperature<span class="ot">=</span>data_mean <span class="sc">%&gt;%</span> </span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> interval_id<span class="sc">/</span><span class="dv">24</span>, <span class="at">y =</span> mean_temp, <span class="at">color =</span> Unit)) <span class="sc">+</span></span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Day"</span>, <span class="at">y =</span> <span class="st">"Average temperature (°C)"</span>, <span class="at">color =</span> <span class="st">"Unit"</span>) <span class="sc">+</span></span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>temperature_color)<span class="sc">+</span></span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span>temperature_color)<span class="sc">+</span></span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>  my_theme </span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>plot_mean_temperature</span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a>plot_mean_humidity<span class="ot">=</span>data_mean <span class="sc">%&gt;%</span> </span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> interval_id<span class="sc">/</span><span class="dv">24</span>, <span class="at">y =</span> mean_humidity, <span class="at">color =</span> Unit)) <span class="sc">+</span></span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Day"</span>, <span class="at">y =</span> <span class="st">"Average humidity (%)"</span>, <span class="at">color =</span> <span class="st">"Unit"</span>) <span class="sc">+</span></span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> scales<span class="sc">::</span><span class="fu">percent_format</span>(<span class="at">scale =</span> <span class="dv">100</span>))<span class="sc">+</span></span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>temperature_color)<span class="sc">+</span></span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span>temperature_color)<span class="sc">+</span></span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a>  my_theme</span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a>plot_mean_humidity</span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-47"><a href="#cb2-47" aria-hidden="true" tabindex="-1"></a>plot_mean_PAR<span class="ot">=</span>data_mean <span class="sc">%&gt;%</span> </span>
<span id="cb2-48"><a href="#cb2-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> interval_id<span class="sc">/</span><span class="dv">24</span>, <span class="at">y =</span> mean_PAR, <span class="at">color =</span> Unit)) <span class="sc">+</span></span>
<span id="cb2-49"><a href="#cb2-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb2-50"><a href="#cb2-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Day"</span>, </span>
<span id="cb2-51"><a href="#cb2-51" aria-hidden="true" tabindex="-1"></a>       <span class="at">y=</span><span class="fu">expression</span>(<span class="fu">atop</span>(<span class="st">"Photosynthetically active"</span>,<span class="fu">paste</span>( <span class="st">"radiation "</span>, <span class="st">"(µmol/ "</span>,m<span class="sc">^-</span><span class="dv">2</span>,<span class="st">" /"</span>,s<span class="sc">^-</span><span class="dv">1</span>,<span class="st">")"</span>))),</span>
<span id="cb2-52"><a href="#cb2-52" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">"Unit"</span>) <span class="sc">+</span></span>
<span id="cb2-53"><a href="#cb2-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>temperature_color)<span class="sc">+</span></span>
<span id="cb2-54"><a href="#cb2-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span>temperature_color)<span class="sc">+</span></span>
<span id="cb2-55"><a href="#cb2-55" aria-hidden="true" tabindex="-1"></a>  my_theme</span>
<span id="cb2-56"><a href="#cb2-56" aria-hidden="true" tabindex="-1"></a>plot_mean_PAR</span>
<span id="cb2-57"><a href="#cb2-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-58"><a href="#cb2-58" aria-hidden="true" tabindex="-1"></a>combined_plot<span class="ot">=</span><span class="fu">ggarrange</span>(plot_mean_temperature,plot_mean_humidity,plot_mean_PAR,<span class="at">ncol=</span><span class="dv">1</span>, <span class="at">nrow=</span><span class="dv">3</span>,<span class="at">common.legend =</span> <span class="cn">TRUE</span>, <span class="at">legend=</span><span class="st">"right"</span>,<span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"A"</span>,<span class="st">"B"</span>,<span class="st">"C"</span>),<span class="at">align=</span><span class="st">'v'</span>)</span>
<span id="cb2-59"><a href="#cb2-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-60"><a href="#cb2-60" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a title and subtitle</span></span>
<span id="cb2-61"><a href="#cb2-61" aria-hidden="true" tabindex="-1"></a>combined_plot_title<span class="ot">=</span><span class="fu">annotate_figure</span>(</span>
<span id="cb2-62"><a href="#cb2-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate_figure</span>(combined_plot,</span>
<span id="cb2-63"><a href="#cb2-63" aria-hidden="true" tabindex="-1"></a>                  <span class="at">top=</span><span class="fu">text_grob</span>(<span class="st">"The measurement represents an average each hour derived from four sensors positioned at </span><span class="sc">\n</span><span class="st"> various locations within the unit for each unit."</span>, <span class="at">size =</span> <span class="dv">12</span>,<span class="at">hjust=</span><span class="dv">0</span>,<span class="at">x=</span><span class="fl">0.1</span>,<span class="at">y=</span><span class="fl">0.4</span>),</span>
<span id="cb2-64"><a href="#cb2-64" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb2-65"><a href="#cb2-65" aria-hidden="true" tabindex="-1"></a>  <span class="at">top=</span><span class="fu">text_grob</span>(<span class="st">"Temperature, humidity and PAR during the experiment"</span>,<span class="at">face =</span> <span class="st">"bold"</span>, <span class="at">size =</span> <span class="dv">14</span>,<span class="at">hjust=</span><span class="dv">0</span>,<span class="at">x=</span><span class="fl">0.1</span>,<span class="at">y=</span><span class="fl">0.4</span>)</span>
<span id="cb2-66"><a href="#cb2-66" aria-hidden="true" tabindex="-1"></a>) ; combined_plot_title</span>
<span id="cb2-67"><a href="#cb2-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-68"><a href="#cb2-68" aria-hidden="true" tabindex="-1"></a><span class="co"># export</span></span>
<span id="cb2-69"><a href="#cb2-69" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"report/intro/plot/combined_plot.svg"</span>), combined_plot_title, <span class="at">width =</span> <span class="dv">21</span>, <span class="at">height =</span> <span class="dv">25</span>, <span class="at">units =</span> <span class="st">"cm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div id="fig-metadata" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-metadata-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="plot/combined_plot.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-3" title="Figure&nbsp;2.3: Temperature (A), humidity (B) and PAR (C) during the experiment. The measurement represents an average each hour derived from four sensors positioned at various locations within the unit for each unit."><img src="plot/combined_plot.svg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-metadata-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2.3: Temperature (A), humidity (B) and PAR (C) during the experiment. The measurement represents an average each hour derived from four sensors positioned at various locations within the unit for each unit.
</figcaption>
</figure>
</div>
<details>
<summary style="color: #007bff;">
Bonnus: Correction for spatial trends over time
</summary>
<section id="correction-for-spatial-trends-over-time" class="level3" data-number="2.2.1">
<h3 data-number="2.2.1" class="anchored" data-anchor-id="correction-for-spatial-trends-over-time"><span class="header-section-number">2.2.1</span> Correction for spatial trends over time</h3>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
If there had been a significant bias, I would have had to take it into account
</div>
</div>
<div class="callout-body-container callout-body">
<p>But according to the model (<a href="../water/2_water_during_the_experiment.html" class="quarto-xref"><span>Chapter 3</span></a> and <a href="../physio/2_rapid_analysis_ecophy.html" class="quarto-xref"><span>Chapter 6</span></a>) using biomass there is only a slight effect of position in the greenhouse, and in addition I randomized the experiments.</p>
</div>
</div>
<section id="data-importation" class="level5" data-number="2.2.1.0.1">
<h5 data-number="2.2.1.0.1" class="anchored" data-anchor-id="data-importation"><span class="header-section-number">2.2.1.0.1</span> Data importation</h5>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>df_physio<span class="ot">&lt;-</span><span class="fu">read.csv2</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/physio/global_physio.csv"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(<span class="st">"genotype"</span>,<span class="st">"water_condition"</span>,<span class="st">"heat_condition"</span>), as.factor)) <span class="sc">%&gt;%</span> </span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(<span class="st">"storage_line"</span>,<span class="st">"position"</span>), as.integer)) <span class="sc">%&gt;%</span> </span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">water_condition=</span><span class="fu">relevel</span>(water_condition, <span class="st">"WW"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">heat_condition=</span><span class="fu">relevel</span>(heat_condition, <span class="st">"OT"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="formatting-and-visualizing-the-data" class="level5" data-number="2.2.1.0.2">
<h5 data-number="2.2.1.0.2" class="anchored" data-anchor-id="formatting-and-visualizing-the-data"><span class="header-section-number">2.2.1.0.2</span> Formatting and visualizing the data</h5>
<p>The first step when modeling platform experiment data with the statgenHTP package is creating an object of class TP (Time Points) <em>For some variable you havent (one harvest)</em>. In this object, the time points are split into single data.frames. It is then used throughout the statgenHTP package as input for analyses.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Formatting the data = create a list with one data frame per time point</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>U2_all_time_point <span class="ot">&lt;-</span> <span class="fu">createTimePoints</span>(<span class="at">dat =</span> df_physio <span class="sc">%&gt;%</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">filter</span>(unit<span class="sc">==</span><span class="st">"U2"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>                                    <span class="co">#filter(paire==F) %&gt;% </span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">mutate</span>(<span class="at">plant_num=</span><span class="fu">as.character</span>(plant_num)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">mutate</span>(<span class="at">position_b =</span><span class="fu">ifelse</span>(<span class="fu">as.numeric</span>(plant_num) <span class="sc">%%</span> <span class="dv">2</span> <span class="sc">==</span> <span class="dv">1</span>, <span class="fu">as.numeric</span>(position) <span class="sc">*</span> <span class="dv">2</span> <span class="sc">-</span> <span class="dv">1</span>, <span class="fu">as.numeric</span>(position) <span class="sc">*</span> <span class="dv">2</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">mutate</span>(<span class="at">pos=</span><span class="fu">paste0</span>(<span class="st">"c"</span>,storage_line,<span class="st">"r"</span>,position_b)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">mutate</span>(<span class="at">timepoints=</span><span class="st">"2021-10-11"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">mutate</span>(<span class="at">harvest_condition=</span><span class="fu">paste0</span>(recolte, <span class="st">"_"</span>, condition)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>                                    dplyr<span class="sc">::</span><span class="fu">select</span>(timepoints, harvest_condition, position_b, storage_line, sum_biomass,pos,rep),</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">experimentName =</span> <span class="st">"U2_Soybean_2021"</span>,</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">genotype =</span> <span class="st">"harvest_condition"</span>,</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">timePoint =</span> <span class="st">"timepoints"</span>,</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">repId =</span> <span class="st">"rep"</span>, </span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">plotId =</span> <span class="st">"pos"</span>,</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">rowNum =</span> <span class="st">"position_b"</span>, <span class="at">colNum =</span> <span class="st">"storage_line"</span>,</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">addCheck =</span> <span class="cn">FALSE</span>, <span class="co">#if you have some geneotype repeate (that help the model to control)</span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">checkGenotypes =</span> <span class="fu">c</span>(<span class="st">"Wen_WW_OT"</span>, <span class="st">"Wen_WW_HS"</span>, <span class="st">"Sto_WW_OT"</span>, <span class="st">"Sto_WW_HS"</span>)</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>                                )</span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>U3_all_time_point <span class="ot">&lt;-</span> <span class="fu">createTimePoints</span>(<span class="at">dat =</span> df_physio <span class="sc">%&gt;%</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">filter</span>(unit<span class="sc">==</span><span class="st">"U3"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>                                    <span class="co">#filter(paire==F) %&gt;% </span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">mutate</span>(<span class="at">plant_num=</span><span class="fu">as.character</span>(plant_num)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">mutate</span>(<span class="at">position_b =</span><span class="fu">ifelse</span>(<span class="fu">as.numeric</span>(plant_num) <span class="sc">%%</span> <span class="dv">2</span> <span class="sc">==</span> <span class="dv">1</span>, <span class="fu">as.numeric</span>(position) <span class="sc">*</span> <span class="dv">2</span> <span class="sc">-</span> <span class="dv">1</span>, <span class="fu">as.numeric</span>(position) <span class="sc">*</span> <span class="dv">2</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">mutate</span>(<span class="at">pos=</span><span class="fu">paste0</span>(<span class="st">"c"</span>,storage_line,<span class="st">"r"</span>,position_b)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">mutate</span>(<span class="at">timepoints=</span><span class="st">"2021-10-11"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">mutate</span>(<span class="at">harvest_condition=</span><span class="fu">paste0</span>(recolte, <span class="st">"_"</span>, condition)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>                                    dplyr<span class="sc">::</span><span class="fu">select</span>(timepoints, harvest_condition, position_b, storage_line, sum_biomass,pos,rep),</span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">experimentName =</span> <span class="st">"U3_Soybean_2021"</span>,</span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">genotype =</span> <span class="st">"harvest_condition"</span>,</span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">timePoint =</span> <span class="st">"timepoints"</span>,</span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">repId =</span> <span class="st">"rep"</span>, </span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">plotId =</span> <span class="st">"pos"</span>,</span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">rowNum =</span> <span class="st">"position_b"</span>, <span class="at">colNum =</span> <span class="st">"storage_line"</span>,</span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">addCheck =</span> <span class="cn">FALSE</span>, <span class="co">#if you have some geneotype repeate (that help the model to control)</span></span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">checkGenotypes =</span> <span class="fu">c</span>(<span class="st">"Wen_WW_OT"</span>, <span class="st">"Wen_WW_HS"</span>, <span class="st">"Sto_WW_OT"</span>, <span class="st">"Sto_WW_HS"</span>)</span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a>                                )</span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a>U2_H2_time_point <span class="ot">&lt;-</span> <span class="fu">createTimePoints</span>(<span class="at">dat =</span> df_physio <span class="sc">%&gt;%</span></span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">filter</span>(unit<span class="sc">==</span><span class="st">"U2"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">filter</span>(paire<span class="sc">==</span>F) <span class="sc">%&gt;%</span> </span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">filter</span>(recolte<span class="sc">==</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">mutate</span>(<span class="at">plant_num=</span><span class="fu">as.character</span>(plant_num)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">mutate</span>(<span class="at">position_b =</span><span class="fu">ifelse</span>(<span class="fu">as.numeric</span>(plant_num) <span class="sc">%%</span> <span class="dv">2</span> <span class="sc">==</span> <span class="dv">1</span>, <span class="fu">as.numeric</span>(position) <span class="sc">*</span> <span class="dv">2</span> <span class="sc">-</span> <span class="dv">1</span>, <span class="fu">as.numeric</span>(position) <span class="sc">*</span> <span class="dv">2</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">mutate</span>(<span class="at">pos=</span><span class="fu">paste0</span>(<span class="st">"c"</span>,storage_line,<span class="st">"r"</span>,position_b)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">mutate</span>(<span class="at">timepoints=</span><span class="st">"2021-10-11"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">mutate</span>(<span class="at">harvest_condition=</span><span class="fu">paste0</span>(recolte, <span class="st">"_"</span>, condition)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-49"><a href="#cb4-49" aria-hidden="true" tabindex="-1"></a>                                    dplyr<span class="sc">::</span><span class="fu">select</span>(timepoints, harvest_condition, position, storage_line, sum_biomass,pos,rep),</span>
<span id="cb4-50"><a href="#cb4-50" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">experimentName =</span> <span class="st">"U2_Soybean_2021"</span>,</span>
<span id="cb4-51"><a href="#cb4-51" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">genotype =</span> <span class="st">"harvest_condition"</span>,</span>
<span id="cb4-52"><a href="#cb4-52" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">timePoint =</span> <span class="st">"timepoints"</span>,</span>
<span id="cb4-53"><a href="#cb4-53" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">repId =</span> <span class="st">"rep"</span>, </span>
<span id="cb4-54"><a href="#cb4-54" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">plotId =</span> <span class="st">"pos"</span>,</span>
<span id="cb4-55"><a href="#cb4-55" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">rowNum =</span> <span class="st">"position"</span>, <span class="at">colNum =</span> <span class="st">"storage_line"</span>,</span>
<span id="cb4-56"><a href="#cb4-56" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">addCheck =</span> <span class="cn">FALSE</span>, <span class="co">#if you have some geneotype repeate (that help the model to control)</span></span>
<span id="cb4-57"><a href="#cb4-57" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">checkGenotypes =</span> <span class="fu">c</span>(<span class="st">"Wen_WW_OT"</span>, <span class="st">"Wen_WW_HS"</span>, <span class="st">"Sto_WW_OT"</span>, <span class="st">"Sto_WW_HS"</span>)</span>
<span id="cb4-58"><a href="#cb4-58" aria-hidden="true" tabindex="-1"></a>                                )</span>
<span id="cb4-59"><a href="#cb4-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-60"><a href="#cb4-60" aria-hidden="true" tabindex="-1"></a>U3_H2_time_point <span class="ot">&lt;-</span> <span class="fu">createTimePoints</span>(<span class="at">dat =</span> df_physio <span class="sc">%&gt;%</span></span>
<span id="cb4-61"><a href="#cb4-61" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">filter</span>(unit<span class="sc">==</span><span class="st">"U3"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-62"><a href="#cb4-62" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">filter</span>(paire<span class="sc">==</span>F) <span class="sc">%&gt;%</span> </span>
<span id="cb4-63"><a href="#cb4-63" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">filter</span>(recolte<span class="sc">==</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-64"><a href="#cb4-64" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">mutate</span>(<span class="at">plant_num=</span><span class="fu">as.character</span>(plant_num)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-65"><a href="#cb4-65" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">mutate</span>(<span class="at">position_b =</span><span class="fu">ifelse</span>(<span class="fu">as.numeric</span>(plant_num) <span class="sc">%%</span> <span class="dv">2</span> <span class="sc">==</span> <span class="dv">1</span>, <span class="fu">as.numeric</span>(position) <span class="sc">*</span> <span class="dv">2</span> <span class="sc">-</span> <span class="dv">1</span>, <span class="fu">as.numeric</span>(position) <span class="sc">*</span> <span class="dv">2</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-66"><a href="#cb4-66" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">mutate</span>(<span class="at">pos=</span><span class="fu">paste0</span>(<span class="st">"c"</span>,storage_line,<span class="st">"r"</span>,position)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-67"><a href="#cb4-67" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">mutate</span>(<span class="at">timepoints=</span><span class="st">"2021-10-11"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-68"><a href="#cb4-68" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">mutate</span>(<span class="at">harvest_condition=</span><span class="fu">paste0</span>(recolte, <span class="st">"_"</span>, condition)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-69"><a href="#cb4-69" aria-hidden="true" tabindex="-1"></a>                                      dplyr<span class="sc">::</span><span class="fu">select</span>(timepoints, harvest_condition, position, storage_line, sum_biomass,pos,rep),</span>
<span id="cb4-70"><a href="#cb4-70" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">experimentName =</span> <span class="st">"U3_Soybean_2021"</span>,</span>
<span id="cb4-71"><a href="#cb4-71" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">genotype =</span> <span class="st">"harvest_condition"</span>,</span>
<span id="cb4-72"><a href="#cb4-72" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">timePoint =</span> <span class="st">"timepoints"</span>,</span>
<span id="cb4-73"><a href="#cb4-73" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">repId =</span> <span class="st">"rep"</span>, </span>
<span id="cb4-74"><a href="#cb4-74" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">plotId =</span> <span class="st">"pos"</span>,</span>
<span id="cb4-75"><a href="#cb4-75" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">rowNum =</span> <span class="st">"position"</span>, <span class="at">colNum =</span> <span class="st">"storage_line"</span>,</span>
<span id="cb4-76"><a href="#cb4-76" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">addCheck =</span> <span class="cn">FALSE</span>, <span class="co">#if you have some geneotype repeate (that help the model to control)</span></span>
<span id="cb4-77"><a href="#cb4-77" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">checkGenotypes =</span> <span class="fu">c</span>(<span class="st">"Wen_WW_OT"</span>, <span class="st">"Wen_WW_HS"</span>, <span class="st">"Sto_WW_OT"</span>, <span class="st">"Sto_WW_HS"</span>)</span>
<span id="cb4-78"><a href="#cb4-78" aria-hidden="true" tabindex="-1"></a>                                )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Plot the layout, NA indicate a probleme with the plant. (It has not been harvested)</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Plot the layout for the 3rd time points.</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>p_u2<span class="ot">=</span><span class="fu">plot</span>(U2_all_time_point,</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>     <span class="at">plotType =</span> <span class="st">"layout"</span>,</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">timePoints =</span> <span class="fu">c</span>(<span class="dv">1</span>),</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">highlight =</span> <span class="fu">c</span>(<span class="st">"1_Wen_WW_OT"</span>, <span class="st">"1_Sto_WW_OT"</span>,<span class="st">"2_Wen_WW_OT"</span>, <span class="st">"2_Wen_WS_OT"</span>, <span class="st">"2_Wen_WW_HS"</span>, <span class="st">"2_Wen_WS_HS"</span>,<span class="st">"2_Sto_WW_OT"</span>, <span class="st">"2_Sto_WS_OT"</span>, <span class="st">"2_Sto_WW_HS"</span>, <span class="st">"2_Sto_WS_HS"</span>),</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">showGeno =</span> <span class="cn">TRUE</span>)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>p_u3<span class="ot">=</span><span class="fu">plot</span>(U3_all_time_point,</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>     <span class="at">plotType =</span> <span class="st">"layout"</span>,</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>     <span class="at">timePoints =</span> <span class="fu">c</span>(<span class="dv">1</span>),</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">highlight =</span> <span class="fu">c</span>(<span class="st">"1_Wen_WW_OT"</span>, <span class="st">"1_Sto_WW_OT"</span>,<span class="st">"2_Wen_WW_OT"</span>, <span class="st">"2_Wen_WS_OT"</span>, <span class="st">"2_Wen_WW_HS"</span>, <span class="st">"2_Wen_WS_HS"</span>,<span class="st">"2_Sto_WW_OT"</span>, <span class="st">"2_Sto_WS_OT"</span>, <span class="st">"2_Sto_WW_HS"</span>, <span class="st">"2_Sto_WS_HS"</span>),</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>     <span class="at">showGeno =</span> <span class="cn">TRUE</span>)</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a><span class="fu">svg</span>(<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"report/intro/plot/layout_unit_U2.svg"</span>),<span class="at">width =</span> <span class="dv">9</span>, <span class="at">height =</span> <span class="dv">9</span>)</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>p_u2</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a><span class="fu">svg</span>(<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"report/intro/plot/layout_unit_U3.svg"</span>),<span class="at">width =</span> <span class="dv">9</span>, <span class="at">height =</span> <span class="dv">9</span>)</span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>p_u3</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a><span class="co"># # Check the heatmap of the raw data at one time point</span></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a><span class="co"># plot(U2_time_point,</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a><span class="co">#      plotType = "layout",</span></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a><span class="co">#      timePoints = 1,</span></span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a><span class="co">#      traits = "sum_biomass")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div id="fig-greenhouse" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-greenhouse-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-greenhouse" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-U2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-U2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="plot/layout_unit_U2.svg" class="lightbox" data-gallery="fig-greenhouse" title="Figure&nbsp;2.4&nbsp;(a): U2"><img src="plot/layout_unit_U2.svg" class="img-fluid figure-img" data-ref-parent="fig-greenhouse"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-U2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) U2
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-greenhouse" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-U3" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-U3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="plot/layout_unit_U3.svg" class="lightbox" data-gallery="fig-greenhouse" title="Figure&nbsp;2.4&nbsp;(b): U3"><img src="plot/layout_unit_U3.svg" class="img-fluid figure-img" data-ref-parent="fig-greenhouse"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-U3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) U3
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-greenhouse-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2.4: Layout of Rhizotube and plants in the two greenhouses
</figcaption>
</figure>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>Trait <span class="ot">&lt;-</span><span class="st">"sum_biomass"</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>spatial_correction<span class="ot">&lt;-</span><span class="cf">function</span>(Trait,data_x){</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  pheno <span class="ot">&lt;-</span> data_x[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span> <span class="fu">drop_na</span>(Trait) <span class="co">#%&gt;% filter(genotype %in% c( "2_Wen_WS_OT","2_Sto_WS_OT")) %&gt;% mutate(genotype==substr(genotype, 3, 5)) # select the nb of segments: here all the rows and columns</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  nseg <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">nlevels</span>(pheno<span class="sc">$</span>colId), <span class="fu">nlevels</span>(pheno<span class="sc">$</span>rowId))</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"Time Point: "</span>, timePoint, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, <span class="at">file =</span> output_file)  <span class="co"># add timePoint number</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(timePoint)</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  fit.SpATS <span class="ot">&lt;-</span> <span class="fu">SpATS</span>(<span class="at">response =</span> Trait, <span class="co"># fit the SpATS model</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>                   <span class="at">fixed =</span> <span class="sc">~</span> repId,</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>                   <span class="at">random =</span> <span class="sc">~</span> colId <span class="sc">+</span> rowId,</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>                   <span class="at">spatial =</span> <span class="sc">~</span> <span class="fu">PSANOVA</span>(colNum, rowNum,</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">nseg =</span> nseg,</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">nest.div =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">2</span>)),</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>                   <span class="at">genotype =</span> <span class="st">"genotype"</span>,</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>                   <span class="at">genotype.as.random =</span> <span class="cn">TRUE</span>,</span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>                   <span class="at">data =</span> pheno,</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>                   <span class="at">control =</span> <span class="fu">list</span>(<span class="at">maxit =</span> <span class="dv">50</span>,</span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">tolerance =</span> <span class="fl">1e-03</span>,</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">monitoring =</span> <span class="dv">0</span>)) </span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>  summary_output <span class="ot">&lt;-</span> <span class="fu">capture.output</span>(<span class="fu">summary</span>(fit.SpATS)) <span class="co"># Captures model summary output</span></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>  all_summaries <span class="ot">&lt;-</span> <span class="fu">c</span>(all_summaries, <span class="fu">paste0</span>(<span class="st">"Time Point: "</span>, timePoint, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, summary_output))<span class="co"># Add output to variable</span></span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">writeLines</span>(all_summaries, <span class="at">con =</span> output_file)</span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"Traitement pour le trait"</span>, Trait, <span class="st">"terminé.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(fit.SpATS, <span class="at">spaTrend =</span> <span class="st">"percentage"</span>)<span class="do">## plot of the spats model with spatial trend as percentage</span></span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a>  modPhenoSpCheck <span class="ot">&lt;-</span> <span class="fu">fitModels</span>(<span class="at">TP =</span> data_x, </span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">trait =</span> Trait,</span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">useRepId =</span> <span class="cn">TRUE</span>,</span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">useCheck =</span> <span class="cn">FALSE</span>,</span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">timePoints =</span> <span class="dv">1</span>)</span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a>  px<span class="ot">&lt;-</span><span class="fu">plot</span>(modPhenoSpCheck, </span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a>           <span class="at">timePoints =</span> <span class="dv">1</span>,</span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a>           <span class="at">plotType =</span> <span class="st">"spatial"</span>,</span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a>           <span class="at">spaTrend =</span> <span class="st">"percentage"</span>)</span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb6-40"><a href="#cb6-40" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Extract the corrected values for one time point</span></span>
<span id="cb6-41"><a href="#cb6-41" aria-hidden="true" tabindex="-1"></a>  spatCorrSp <span class="ot">&lt;-</span> <span class="fu">getCorrected</span>(modPhenoSpCheck, <span class="at">timePoints =</span> timePoint)</span>
<span id="cb6-42"><a href="#cb6-42" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb6-43"><a href="#cb6-43" aria-hidden="true" tabindex="-1"></a>  varianceSp <span class="ot">&lt;-</span> <span class="fu">getVar</span>(modPhenoSpCheck) <span class="do">## Extract model components: </span></span>
<span id="cb6-44"><a href="#cb6-44" aria-hidden="true" tabindex="-1"></a>  heritSp    <span class="ot">&lt;-</span> <span class="fu">getHerit</span>(modPhenoSpCheck)</span>
<span id="cb6-45"><a href="#cb6-45" aria-hidden="true" tabindex="-1"></a>  effDimSp   <span class="ot">&lt;-</span> <span class="fu">getEffDims</span>(modPhenoSpCheck)</span>
<span id="cb6-46"><a href="#cb6-46" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-47"><a href="#cb6-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(<span class="fu">list</span>(<span class="at">plot =</span> px, <span class="at">datacorrect =</span> spatCorrSp)))</span>
<span id="cb6-48"><a href="#cb6-48" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-49"><a href="#cb6-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-50"><a href="#cb6-50" aria-hidden="true" tabindex="-1"></a>result_U2<span class="ot">=</span><span class="fu">spatial_correction</span>(<span class="at">Trait =</span> <span class="st">"sum_biomass"</span>,<span class="at">data_x =</span> U2_H2_time_point)</span>
<span id="cb6-51"><a href="#cb6-51" aria-hidden="true" tabindex="-1"></a>result_U3<span class="ot">=</span><span class="fu">spatial_correction</span>(<span class="at">Trait =</span> <span class="st">"sum_biomass"</span>,<span class="at">data_x =</span> U3_H2_time_point)</span>
<span id="cb6-52"><a href="#cb6-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-53"><a href="#cb6-53" aria-hidden="true" tabindex="-1"></a>p1<span class="ot">&lt;-</span>result_U2[[<span class="dv">1</span>]]<span class="sc">$</span>plot<span class="sc">$</span><span class="st">`</span><span class="at">2021-10-11</span><span class="st">`</span><span class="sc">$</span>p4</span>
<span id="cb6-54"><a href="#cb6-54" aria-hidden="true" tabindex="-1"></a>p2<span class="ot">&lt;-</span>result_U3[[<span class="dv">1</span>]]<span class="sc">$</span>plot<span class="sc">$</span><span class="st">`</span><span class="at">2021-10-11</span><span class="st">`</span><span class="sc">$</span>p4</span>
<span id="cb6-55"><a href="#cb6-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-56"><a href="#cb6-56" aria-hidden="true" tabindex="-1"></a>p1<span class="ot">=</span>p1<span class="sc">+</span><span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste0</span>(<span class="st">"Filtted Spatial Trend Unit 2"</span>, <span class="st">" "</span>, <span class="st">"sum_biomass"</span>))</span>
<span id="cb6-57"><a href="#cb6-57" aria-hidden="true" tabindex="-1"></a>p2<span class="ot">=</span>p2<span class="sc">+</span><span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste0</span>(<span class="st">"Filtted Spatial Trend Unit 3"</span> , <span class="st">" "</span>, <span class="st">"sum_biomass"</span>))</span>
<span id="cb6-58"><a href="#cb6-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-59"><a href="#cb6-59" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"report/intro/plot/ggcoef_model.svg"</span>),p1<span class="sc">+</span>p2)</span>
<span id="cb6-60"><a href="#cb6-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-61"><a href="#cb6-61" aria-hidden="true" tabindex="-1"></a>result_all<span class="ot">=</span><span class="fu">rbind</span>(result_U2[[<span class="dv">1</span>]]<span class="sc">$</span><span class="st">"datacorrect"</span>,result_U3[[<span class="dv">1</span>]]<span class="sc">$</span><span class="st">"datacorrect"</span>)</span>
<span id="cb6-62"><a href="#cb6-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-63"><a href="#cb6-63" aria-hidden="true" tabindex="-1"></a>p1<span class="ot">=</span>result_all <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>genotype,<span class="at">y=</span>sum_biomass, <span class="at">fill=</span>genotype))<span class="sc">+</span><span class="fu">geom_boxplot</span>()<span class="sc">+</span><span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Without correction for sum biomass"</span>)</span>
<span id="cb6-64"><a href="#cb6-64" aria-hidden="true" tabindex="-1"></a>p2<span class="ot">=</span>result_all <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>genotype,<span class="at">y=</span>sum_biomass_corr, <span class="at">fill=</span>genotype))<span class="sc">+</span><span class="fu">geom_boxplot</span>()<span class="sc">+</span><span class="fu">labs</span>(<span class="at">title=</span><span class="st">"With correction for sum biomass"</span>)</span>
<span id="cb6-65"><a href="#cb6-65" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"report/intro/plot/comparison.svg"</span>),p1<span class="sc">+</span>p2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><a href="plot/ggcoef_model.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-6"><img src="plot/ggcoef_model.svg" class="img-fluid"></a></p>
<p><a href="plot/comparison.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-7"><img src="plot/comparison.svg" class="img-fluid"></a></p>
</section></section></details>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list" style="display: none">
<div id="ref-maslard2021" class="csl-entry" role="listitem">
Maslard, Corentin, Mustapha Arkoun, Christophe Salon, and Marion Prudent. 2021. <span>“Root Architecture Characterization in Relation to Biomass Allocation and Biological Nitrogen Fixation in a Collection of European Soybean Genotypes.”</span> <em>OCL</em> 28: 48. <a href="https://doi.org/10.1051/ocl/2021033">https://doi.org/10.1051/ocl/2021033</a>.
</div>
<div id="ref-voisin_symbiotic_2003" class="csl-entry" role="listitem">
Voisin, A. S., Salon, Christophe, Jeudy, Christian, and F. R. Warembourg. 2003. <span>“Symbiotic N2 Fixation Activity in Relation to c Economy of Pisum Sativum l. As a Function of Plant Phenology.”</span> <em>Journal of Experimental Botany</em> 54 (393): 2733–44. <a href="https://doi.org/10.1093/jxb/erg290">https://doi.org/10.1093/jxb/erg290</a>.
</div>
</div>
</section>



</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../../report/intro/intro.html" class="pagination-link" aria-label="Introduction">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../../report/water/2_water_during_the_experiment.html" class="pagination-link" aria-label="Use of water during the experiment">
        <span class="nav-page-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Use of water during the experiment</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://creativecommons.org/licenses/by-nc-sa/4.0/">
      <i class="bi bi-badge-cc-fill" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item">
    <a class="nav-link" href="https://cmaslard.xyz/">
<p>Corentin Maslard 2024</p>
</a>
  </li>  
</ul>
    </div>   
    <div class="nav-footer-center">
      <ul class="footer-items list-unstyled">
    <li class="nav-item">
    <a class="nav-link" href="mailto:maslard.corentin@gmail.com">
<p><iconify-icon role="img" inline="" icon="fa6-solid:envelope" aria-label="Icon envelope from fa6-solid Iconify.design set." title="Icon envelope from fa6-solid Iconify.design set."></iconify-icon></p>
</a>
  </li>  
    <li class="nav-item">
 
  </li>  
    <li class="nav-item compact">
    <a class="nav-link active" href="https://github.com/cmaslard/phd_soybean_2021_2024/" aria-current="page">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item">
    <a class="nav-link" href="https://www.linkedin.com/in/corentin-maslard">
<p><iconify-icon role="img" inline="" icon="fa6-brands:linkedin" aria-label="Icon linkedin from fa6-brands Iconify.design set." title="Icon linkedin from fa6-brands Iconify.design set."></iconify-icon></p>
</a>
  </li>  
    <li class="nav-item">
    <a class="nav-link" href="https://x.com/cmaslard">
<p><iconify-icon role="img" inline="" icon="fa6-brands:square-x-twitter" aria-label="Icon square-x-twitter from fa6-brands Iconify.design set." title="Icon square-x-twitter from fa6-brands Iconify.design set."></iconify-icon></p>
</a>
  </li>  
    <li class="nav-item">
    <a class="nav-link" href="https://orcid.org/0000-0002-6951-1963">
<p><iconify-icon role="img" inline="" icon="fa6-brands:orcid" aria-label="Icon orcid from fa6-brands Iconify.design set." title="Icon orcid from fa6-brands Iconify.design set."></iconify-icon></p>
</a>
  </li>  
    <li class="nav-item">
    <a class="nav-link" href="https://www.researchgate.net/profile/Corentin-Maslard">
<p><iconify-icon role="img" inline="" icon="fa6-brands:researchgate" aria-label="Icon researchgate from fa6-brands Iconify.design set." title="Icon researchgate from fa6-brands Iconify.design set."></iconify-icon></p>
</a>
  </li>  
    <li class="nav-item">
    <a class="nav-link" href="https://scholar.google.com/citations?user=TsztyiMAAAAJ&amp;hl" target="blank">
<p><iconify-icon role="img" inline="" icon="academicons:google-scholar" aria-label="Icon google-scholar from academicons Iconify.design set." title="Icon google-scholar from academicons Iconify.design set."></iconify-icon></p>
</a>
  </li>  
    <li class="nav-item">
    <a class="nav-link" href="https://cv.hal.science/corentinmaslard">
<p><iconify-icon role="img" inline="" icon="academicons:hal" aria-label="Icon hal from academicons Iconify.design set." title="Icon hal from academicons Iconify.design set."></iconify-icon></p>
</a>
  </li>  
    <li class="nav-item">
    <a class="nav-link" href="https://docs.google.com/document/d/1oCgrdjnAJuVjgocx3F2kmlbJxlqa6uWrSgqm5fWRhpA/edit?usp=sharing">
<p><iconify-icon role="img" inline="" icon="pepicons-pop:cv" aria-label="Icon cv from pepicons-pop Iconify.design set." title="Icon cv from pepicons-pop Iconify.design set."></iconify-icon></p>
</a>
  </li>  
    <li class="nav-item">
    <a class="nav-link" href="https://1drv.ms/b/s!Aq9vs33-G0z9wfsFkiGfdJDcvK2Ouw?e=uW3vYb">
<p>Draft Thesis Manuscript</p>
</a>
  </li>  
    <li class="nav-item">
    <a class="nav-link" href="https://corentin-maslard.shinyapps.io/shiny_XP1_Corentin_Maslard/">
<p>R-shiny</p>
</a>
  </li>  
</ul>
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/cmaslard/phd_soybean_2021_2024/blob/main/report/intro/1_experimental_design_climat.qmd" class="toc-action"><i class="bi bi-github"></i>View source</a></li><li><a href="https://github.com/cmaslard/phd_soybean_2021_2024/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p><span class="faux-block">Made with <iconify-icon icon="line-md:heart-twotone"></iconify-icon>, <iconify-icon icon="line-md:coffee-twotone-loop"></iconify-icon>, <iconify-icon role="img" inline="" icon="fa6-brands:r-project" aria-label="Icon r-project from fa6-brands Iconify.design set." title="Icon r-project from fa6-brands Iconify.design set."></iconify-icon> and <a href="https://quarto.org/">Quarto</a></span> | <span class="faux-block"><a href="https://github.com/cmaslard/cmaslard.github.io">View the source at <iconify-icon role="img" inline="" icon="fa6-brands:github" aria-label="Icon github from fa6-brands Iconify.design set." title="Icon github from fa6-brands Iconify.design set."></iconify-icon> GitHub</a></span></p>
</div>
  </div>
</footer>
<script>var lightboxQuarto = GLightbox({"descPosition":"right","selector":".lightbox","loop":false,"closeEffect":"fade","openEffect":"fade"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>




</body></html>